### PR TITLE
CORGI-824 exclude all test directory from syft scan

### PR DESCRIPTION
**Pull request by jasinner**
Syft is detecting components incorrectly from test data. I adjusted our existing exclude filter to filter out all directories with 'test' in the path. That covered 2 existing excludes as well as this new one with the rpmlint source code.

I will write a script to delete the existing bad components from rpmlint RPMs and attach it to CORGI-824

### CODE REVIEW

**Comment by jasinner** [comment]
@RedHatProductSecurity/corgi-devs please review

File: corgi/core/migrations/0095_clean_rpmlint_invalid_syft_deps.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from django.db import migrations
+
+
+def clean_rpmlint_invalid_syft_deps(apps, schema_editor):
+    Component = apps.get_model("core", "Component")
+
+    for rpmlint_src in Component.objects.filter(name="rpmlint", arch="src"):
+        # Delete any provide which wasn't detected by koji.listRPMs (the noarch rpm only)
+        rpmlint_src.provides.exclude(meta_attr__source__contains="koji.listRPMs").delete()
```
Comment (juspence) [inline corgi/core/migrations/0095_clean_rpmlint_invalid_syft_deps.py:9]: Should we schedule / rerun the SCA task for these SRPMs after we delete the non-Koji / SCA provides?

File: corgi/core/migrations/0095_clean_rpmlint_invalid_syft_deps.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from django.db import migrations
+
+
+def clean_rpmlint_invalid_syft_deps(apps, schema_editor):
+    Component = apps.get_model("core", "Component")
+
+    for rpmlint_src in Component.objects.filter(name="rpmlint", arch="src"):
```
Comment (juspence) [inline corgi/core/migrations/0095_clean_rpmlint_invalid_syft_deps.py]:
```suggestion
    for rpmlint_src in Component.objects.filter(name="rpmlint", type="RPM", arch="src"):
```

**Review by juspence — APPROVED** [review]
LGTM, comments are minor and sorry for the delay in getting to reviews lately.

File: corgi/core/migrations/0095_clean_rpmlint_invalid_syft_deps.py (hunk 1 above)
Comment (jasinner) [reply corgi/core/migrations/0095_clean_rpmlint_invalid_syft_deps.py:9]: No need. It won't pick up anything new.

**Pull request merged**